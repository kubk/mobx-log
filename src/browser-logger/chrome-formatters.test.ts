import { observable } from 'mobx';
import {
  ArrayFormatter, MapFormatter,
  ObjectFormatter,
  SetFormatter
} from './chrome-formatters';

describe('ChromeFormatters', () => {
  it('ObjectFormatter returns null for non observable object', () => {
    const objectFormatter = new ObjectFormatter();
    expect(objectFormatter.header([1, 2, 3])).toBeNull();
    expect(objectFormatter.header({ a: 1, b: 2 })).toBeNull();
  });

  it('ObjectFormatter formats object', () => {
    const userState = observable({ age: 23, name: 'Test' });

    const objectFormatter = new ObjectFormatter();

    expect(objectFormatter.header(userState)).toMatchSnapshot();
    expect(objectFormatter.body(userState)).toMatchSnapshot();
    expect(objectFormatter.hasBody(userState)).toBeTruthy();
  });

  it('ArrayFormatter returns null for non observable object', () => {
    const arrayFormatter = new ArrayFormatter();
    expect(arrayFormatter.header([1, 2, 3])).toBeNull();
    // @ts-expect-error
    expect(arrayFormatter.header({ a: 1, b: 2 })).toBeNull();
  });

  it('ArrayFormatter formats observable array', () => {
    const users = [
      observable({ age: 23, name: 'Test 1' }),
      observable({ age: 55, name: 'Test 2' }),
      observable({ age: 99, name: 'Test 2' }),
    ];

    const arrayFormatter = new ArrayFormatter();

    expect(arrayFormatter.header(users)).toMatchSnapshot();
    expect(arrayFormatter.body(users)).toMatchSnapshot();
    expect(arrayFormatter.hasBody(users)).toBeTruthy();
  });

  it('MapFormatter returns null for non observable map', () => {
    const mapFormatter = new MapFormatter();
    // @ts-expect-error
    expect(mapFormatter.header([1, 2, 3])).toBeNull();
    // @ts-expect-error
    expect(mapFormatter.header({ a: 1, b: 2 })).toBeNull();
  });

  it('MapFormatter formats observable map', () => {
    const map = observable.map({ a: 1, b: 2 });

    const mapFormatter = new MapFormatter();

    expect(mapFormatter.header(map)).toMatchSnapshot();
    expect(mapFormatter.body(map)).toMatchSnapshot();
    expect(mapFormatter.hasBody(map)).toBeTruthy();
  });

  it('SetFormatter returns null for non observable set', () => {
    const setFormatter = new SetFormatter();
    // @ts-expect-error
    expect(setFormatter.header([1, 2, 3])).toBeNull();
    // @ts-expect-error
    expect(setFormatter.header({ a: 1, b: 2 })).toBeNull();
  });

  it('SetFormatter formats observable set', () => {
    const set = observable.set([1, 2, 3]);

    const setFormatter = new SetFormatter();

    expect(setFormatter.header(set)).toMatchSnapshot();
    expect(setFormatter.body(set)).toMatchSnapshot();
    expect(setFormatter.hasBody(set)).toBeTruthy();
  });
});
