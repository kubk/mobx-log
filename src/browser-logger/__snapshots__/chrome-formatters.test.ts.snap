// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChromeFormatters > ArrayFormatter formats observable array 1`] = `null`;

exports[`ChromeFormatters > ArrayFormatter formats observable array 2`] = `
[
  "ol",
  {
    "style": "list-style-type: none; padding: 0; margin: 0 0 0 12px; font-style: normal",
  },
  [
    "li",
    [
      "object",
      {
        "object": {
          "age": 23,
          "name": "Test 1",
        },
      },
    ],
  ],
  [
    "li",
    [
      "object",
      {
        "object": {
          "age": 55,
          "name": "Test 2",
        },
      },
    ],
  ],
  [
    "li",
    [
      "object",
      {
        "object": {
          "age": 99,
          "name": "Test 2",
        },
      },
    ],
  ],
]
`;

exports[`ChromeFormatters > MapFormatter formats observable map 1`] = `
[
  "span",
  [
    "span",
    {
      "style": "color: rgb(232,98,0)",
    },
    "Map",
  ],
  [
    "span",
    "[2]",
  ],
]
`;

exports[`ChromeFormatters > MapFormatter formats observable map 2`] = `
[
  "ol",
  {
    "style": "list-style-type: none; padding: 0; margin: 0 0 0 12px; font-style: normal",
  },
  [
    "li",
    {},
    [
      "object",
      {
        "object": "a",
      },
    ],
    ": ",
    [
      "object",
      {
        "object": 1,
      },
    ],
  ],
  [
    "li",
    {},
    [
      "object",
      {
        "object": "b",
      },
    ],
    ": ",
    [
      "object",
      {
        "object": 2,
      },
    ],
  ],
]
`;

exports[`ChromeFormatters > ObjectFormatter formats object with computeds 1`] = `
[
  "span",
  [
    "span",
    {
      "style": "color: rgb(232,98,0)",
    },
    "Object",
  ],
  [
    "span",
    "[2]",
  ],
]
`;

exports[`ChromeFormatters > ObjectFormatter formats object with computeds 2`] = `
[
  "ol",
  {
    "style": "list-style-type: none; padding: 0; margin: 0 0 0 12px; font-style: normal",
  },
  [
    "li",
    {},
    [
      "object",
      {
        "object": "age",
      },
    ],
    ": ",
    [
      "object",
      {
        "object": 23,
      },
    ],
  ],
  [
    "li",
    {},
    [
      "object",
      {
        "object": "name",
      },
    ],
    ": ",
    [
      "object",
      {
        "object": "Test",
      },
    ],
  ],
  [
    "li",
    {},
    [
      "object",
      {
        "object": "isAllowed",
      },
    ],
    ": ",
    [
      "object",
      {
        "object": true,
      },
    ],
  ],
  [
    "li",
    {},
    [
      "object",
      {
        "object": "json",
      },
    ],
    ": ",
    [
      "object",
      {
        "object": {
          "age": 23,
          "name": "Test",
        },
      },
    ],
  ],
]
`;

exports[`ChromeFormatters > SetFormatter formats observable set 1`] = `
[
  "span",
  [
    "span",
    {
      "style": "color: rgb(232,98,0)",
    },
    "Set",
  ],
  [
    "span",
    "[3]",
  ],
]
`;

exports[`ChromeFormatters > SetFormatter formats observable set 2`] = `
[
  "ol",
  {
    "style": "list-style-type: none; padding: 0; margin: 0 0 0 12px; font-style: normal",
  },
  [
    "li",
    [
      "object",
      {
        "object": 1,
      },
    ],
  ],
  [
    "li",
    [
      "object",
      {
        "object": 2,
      },
    ],
  ],
  [
    "li",
    [
      "object",
      {
        "object": 3,
      },
    ],
  ],
]
`;
